src/effects/effectsmanager.cpp

```diff
@@ -253,9 +313,17 @@ void EffectsManager::saveEffectsXml() {
         standardEffectChainPresets.append(pPreset);
     }
 
+    auto outputChainPreset = m_outputEffectChain.data() != nullptr
+            ? EffectChainPresetPointer::create(m_outputEffectChain.data())
+            // required for tests
+            : EffectChainPresetPointer(new EffectChainPreset());
```

Comment:
yeah :roll_eyes:

Reply:
yes and no.
no because your proposal returns either a `EffectChainPresetPointer` or a `OutputEffectChainPointer`. we want a preset pointer 

I'm really bad with pointer types and didn't check the docs, I just copied from the loops above (standard and QuickEffect chains), hence I have no idea how to avoid `data()` here.
[`EffectChainPreset::EffectChainPreset(const EffectChain* chain)`](https://github.com/mixxxdj/mixxx/blob/a89f6769c613c59ed1040be5e5a7549465f2d8e2/src/effects/presets/effectchainpreset.cpp#L52-L60) does not alter or delete the pointer, does this mean this concern
> or pass it to another function that could delete it

does not apply here?

Reply:
Oh, I misread the original code. In this case its totally fine. The remaining question then is only whether its fine whether outputChainPreset in the test case should be a nullptr or a default constructed preset.